### PR TITLE
fix: broken link on icons readme page in patternhub

### LINKED ISSUE #2930: more information link doesn't work
### Which generators are impacted?

- [X] All
- [ ] HTML
- [ ] React
- [ ] Angular
- [ ] Vue
- [ ] Web components
- [ ] Power Apps

### Reproduction case

- Open https://db-ui.github.io/mono/version/v0.2.5/foundations/icons/readme
- The link "More information" links to a 404 page

### Expected Behaviour

"More information" would link to an existing page.

### Screenshots

_No response_

### Browser version

None

### Add any other context about the problem here.

_No response_

### PR DESCRIPTION
**Pull request by bruno-sch**
## Proposed changes

Link is corrected. Fixes #2930

## Types of changes

-   [x] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Refactoring (fix on existing components or architectural decisions)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation Update (if none of the other choices apply)

### CODE REVIEW

**Comment by mfranzke** [comment]
@nmerget even though that @bruno-sch has a valid solution here, it might not be the best, as I would assume that we should link to an `.md` file instead of the resulting path.

**Comment by nmerget** [comment]
If we want to enable the link in GitHub and Patternhub, we can't use relative paths. They have to be absolute :(

File: packages/foundations/docs/Icons.md [hunk 1]
```diff
@@ -23,4 +23,4 @@ You can add an icon before or after a tag, by adding an `data-` attribute to you
 
 If you have custom icons and want to use them for foundations and/or in components, you need to generate a **woff2** file.
 
-[More information](./CustomIcons.md)
+[More information](./custom-icons)
```
[the remark this answers is not in the document]
Comment (bruno-sch) [reply packages/foundations/docs/Icons.md:26]: @mfranzke and @nmerget 

That's basically true. In the end, this also applies to all other internal links here: If links point to paths with the file extension `.md`, they work in Github, **but not** in Patternhub. And vice versa.

A possible solution would be to have links in the md files end with the appendix `.md`. However, when copying these files to Patternhub via node-script, remove `.md` from links.

What do you think?

[the remark this answers is not in the document]
Comment (mfranzke) [reply packages/foundations/docs/Icons.md:26]: > @mfranzke and @nmerget
> 
> That's basically true. In the end, this also applies to all other internal links here: If links point to paths with the file extension `.md`, they work in Github, **but not** in Patternhub. And vice versa.
> 
> A possible solution would be to have links in the md files end with the appendix `.md`. However, when copying these files to Patternhub via node-script, remove `.md` from links.
> 
> What do you think?

In either case we would need some kind of transformation of these links to another format. And as the codebase out of GitHub is displayed at github.com out of the box, but does get some kind of transformation for being displayed in patternhub, we should rewrite the URL in this step / setup.

[the remark this answers is not in the document]
Comment (mfranzke) [reply packages/foundations/docs/Icons.md:26]: as aligned with @bruno-sch: the links within markdown need to be correct and during the transformation for GitHub Pages those links would need to get transformed as well to the resulting structure.

**Comment by mfranzke** [comment]
@nmerget could you please have a look how we could set the links within markdown to target other Markdown files and still have those links adapted within the transformation of patternhub ?

**Review by nmerget — APPROVED** [review]

File: packages/foundations/docs/Icons.md [hunk 2]
```diff
@@ -28,4 +28,4 @@ You could use the CSS Custom Property `--db-icon-color` to overwrite the icons c
 
 If you have custom icons and want to use them for foundations and/or in components, you need to generate a **woff2** file.
 
-[More information](./CustomIcons.md)
+[More information](https://github.com/db-ui/mono/blob/main/packages/foundations/docs/CustomIcons.md)
```
Comment (mfranzke) [inline packages/foundations/docs/Icons.md:31]: we would need a solution that does not only do the transform of markdown to HTML regarding the files themselves, but even also the included links. Providing an absolute URL kicks the user out of their context.

Comment (nmerget) [reply packages/foundations/docs/Icons.md:31]: right now we don't have a "good" solution for documentation, it's spread across different sites

For now this should solve the problem

Comment (mfranzke) [reply packages/foundations/docs/Icons.md:31]: https://github.com/db-ui/mono/issues/3614

**Review by mfranzke — APPROVED** [review]

**Pull request merged**